src/misc_gui.cpp

```diff
@@ -1168,6 +1168,7 @@ struct QueryWindow : public Window {
 		this->parent = parent;
 		this->left = parent->left + (parent->width / 2) - (this->width / 2);
 		this->top = parent->top + (parent->height / 2) - (this->height / 2);
+		this->SetDirty();
```

Comment:
Wouldn't it be better to override FindWindowPlacementAndResize, so it can set the location and dirty an area only once?

Reply:
Thanks, I didn't twig that it was overrideable!